src/components/VmModals/DeleteConfirmationModal.js

```diff
@@ -1,36 +1,54 @@
 import React from 'react'
 import PropsTypes from 'prop-types'
-import { Modal } from 'patternfly-react'
+import { MessageDialog, Icon } from 'patternfly-react'
 import { msg } from '../../intl'
 
-const DeleteConfirmationModal = ({ show, onDelete, children, onClose }) => {
-  return (
-    <Modal onHide={onClose} show={show}>
-      <Modal.Header>
-        <button
-          className='close'
-          onClick={onClose}
-        >
-          <span className='pficon pficon-close' title='Close' />
-        </button>
-        <Modal.Title>{msg.confirmDelete()}</Modal.Title>
-      </Modal.Header>
-      <Modal.Body>
-        { children }
-      </Modal.Body>
-      <Modal.Footer>
-        <button className='btn btn-default' onClick={onClose}>{msg.cancel()}</button>
-        <button className='btn btn-danger' onClick={onDelete}>{msg.delete()}</button>
-      </Modal.Footer>
-    </Modal>
-  )
+class DeleteConfirmationModal extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = { show: false }
+    this.handleClose = this.handleClose.bind(this)
+    this.handleDelete = this.handleDelete.bind(this)
+  }
+
+  handleClose () {
+    this.setState({ show: false })
+    this.props.onClose && this.props.onClose()
+  }
+
+  handleDelete () {
+    this.props.onDelete()
+    this.setState({ show: false })
+  }
+
+  render () {
+    const { children, trigger } = this.props
+
+    const icon = <Icon type='pf' name='warning-triangle-o' />
+    return (
+      <React.Fragment>
+        {React.cloneElement(trigger, { onClick: () => this.setState({ show: true }) })}
+        <MessageDialog
+          show={this.state.show}
+          onHide={this.handleClose}
+          primaryAction={this.handleDelete}
+          secondaryAction={this.handleClose}
+          primaryActionButtonContent={msg.delete()}
+          secondaryActionButtonContent={msg.cancel()}
+          title={msg.confirmDelete()}
+          icon={icon}
+          primaryContent={<p className='lead'>{ children }</p>}
+        />
+      </React.Fragment>
+    )
+  }
 }
 
 DeleteConfirmationModal.propTypes = {
-  show: PropsTypes.bool.isRequired,
+  trigger: PropsTypes.node.isRequired,
```

Comment:
This style (also on `RestoreConfirmationModal`) seems a bit awkward, with a trigger component like this.  But I'm not sure how you could change it around to be more like `OverlayTrigger`.

I'm ok with how it is, but it would be worth looking at how it could look a bit more like `OverlayTrigger` (https://react-bootstrap.github.io/components/popovers/#popovers-overlay-trigger)